### ISSUE TITLE
SyntaxNodeExtensions.ReplaceNodes nullable ref annotation should allow null

### ISSUE THREAD

**Issue by AArnott** [dotnet/roslyn#65537]
**Version Used**: 4.4.0

The `SyntaxNodeExtensions` class declares this method:
```cs
        public static TRoot ReplaceNodes<TRoot, TNode>(this TRoot root, IEnumerable<TNode> nodes, Func<TNode, TNode, SyntaxNode> computeReplacementNode)
            where TRoot : SyntaxNode
            where TNode : SyntaxNode
```

The signature forbids `null` to be returned from the `computeReplacementNode` function, although I have code that does that and it works fine. This creates a false CS8603 warning when my delegate returns `null`.
